recipes/pcapplusplus/all/conanfile.py

```diff
@@ -70,55 +72,68 @@ def _configure_sh_script(self):
 
     def _patch_sources(self):
         if not self.options.get_safe("fPIC") and self.settings.os != "Windows":
-            files.replace_in_file(self, os.path.join(self._source_subfolder, "PcapPlusPlus.mk.common"),
+            replace_in_file(self, os.path.join(self.source_folder, "PcapPlusPlus.mk.common"),
                                   "-fPIC", "")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            files.patch(self, **patch)
+        apply_conandata_patches(self)
+        if self.settings.os != "Windows":
+            rename(self, os.path.join(self.source_folder, self._configure_sh_script), os.path.join(self.source_folder, "configure"))
+        else:
+            props_file = os.path.join(self.generators_folder, "conantoolchain.props")
+            vcxproj_templates = glob.glob(f"{self.source_folder}/mk/vs/*.vcxproj.template")
```

Comment:
this works, but it may be more convenient to use a different layout so that we can use a relative path